lib/serverengine/process_manager.rb

```diff
@@ -204,14 +219,26 @@ def tick(blocking_timeout=0)
         return nil
       end
 
-      ready_pipes, _, _ = IO.select(@rpipes.keys, nil, nil, blocking_timeout)
+      if ServerEngine.windows?
+        # TODO: should use WaitForMultipleObjects?
+        ready_pipes = @rpipes.keys
```

Comment:
Because of semantic differences of pipes, this code could be tricky:

* On UNIX, ProcessManager detects exist of a child process using reading from rpipe. If reading from rpipe failed with EOFError, it means that the child process exits. That's why ServerEngine can safely ignore SIGCHLD.
* But on Windows, Win32::Pipe::Server doesn't cause EOFError even when a child process exits (if my understanding is correct). If enable_heartbeat is true, ProcessManager still can know exit of a child process. However, if it is false, ProcessManager needs to detect exit of a child process in different way.

An idea is, as you mention here in the comment, using WaitForMultipleObjects to wait for both pipe and exits of child processes.
Another idea is throw an exception at `#initialize` if enable_heartbeat is false on Windows.